ddtrace/internal/telemetry/writer.py

```diff
@@ -86,12 +87,15 @@ def send_event(self, request):
         except Exception:
             log.debug("failed to send telemetry to the Datadog Agent at %s.", self.url, exc_info=True)
         finally:
-            conn.close()
+            try:
+                conn.close()
+            except UnboundLocalError:
```

Comment:
Why are we handling this exception? This seems like a bug

Reply:
It's because in the code above, we can end up not initializing a `conn` object, before we error out. If that's the case, then I think we'd want to catch the exception rather than crash.

Reply:
Yes if the agent url is invalid this will raise a ValueError and `conn` will be uninitialized. I guess we can initialize `conn to None` before the try-except block. Then so something like this:

```
if conn:
    conn.close()
```

 I think this is cleaner than catching `UnboundLocalError`.